rollbar/__init__.py

```diff
@@ -807,48 +799,35 @@ def _add_locals_data(data, exc_info):
             #       return x + a + b + z
             #
             #   func((1, (1, 2), 3))
-            named_args = _flatten_nested_lists(arginfo.args)
-
-            # Fill in all of the named args
-            for named_arg in named_args:
-                if named_arg in local_vars:
-                    args.append(local_vars[named_arg])
+            argspec.extend(_flatten_nested_lists(arginfo.args))
 
-            # Add any varargs
+            # Remove varargs if desired
             if arginfo.varargs is not None:
-                args.extend(local_vars[arginfo.varargs])
+                if SETTINGS['locals']['include_varargs']:
+                    varargspec = arginfo.varargs
+                else:
+                    del local_vars[arginfo.varargs]
 
             # Fill in all of the kwargs
             if arginfo.keywords is not None:
-                kw.update(local_vars[arginfo.keywords])
-
-            if argspec and argspec.defaults:
-                # Put any of the args that have defaults into kwargs
-                num_defaults = len(argspec.defaults)
-                if num_defaults:
-                    # The last len(argspec.defaults) args in arginfo.args should be added
-                    # to kwargs and removed from args
-                    kw.update(dict(zip(arginfo.args[-num_defaults:], args[-num_defaults:])))
-                    args = args[:-num_defaults]
+                keywordspec = arginfo.keywords
 
             # Optionally fill in locals for this frame
             if local_vars and _check_add_locals(cur_frame, frame_num, num_frames):
                 _locals.update(local_vars.items())
```

Comment:
Seems like we should do the check to see if we want to add any locals before any of the argspec stuff above.